src/feature/relay/selftest.c

```diff
@@ -281,20 +281,28 @@ inform_testing_reachability(void)
   control_event_server_status(LOG_NOTICE,
                               "CHECKING_REACHABILITY ORADDRESS=%s:%d",
                               address, me->or_port);
+  const bool has_ipv6 = tor_addr_port_is_valid(&me->ipv6_addr,
+                                               me->ipv6_orport, 0);
+  if (has_ipv6) {
+    tor_snprintf(extra_addrbuf, sizeof(extra_addrbuf), " and %s",
+                 fmt_addrport(&me->ipv6_addr, me->ipv6_orport));
+    /* We'll add an IPv6 control event in #34068. */
+  }
   if (me->dir_port) {
-    tor_snprintf(dirbuf, sizeof(dirbuf), " and DirPort %s:%d",
+    tor_snprintf(extra_addrbuf, sizeof(extra_addrbuf), " and DirPort %s:%d",
```

Comment:
This printf and the one above are bouth writing into extra_addrbuf. I think this one will overwrite the previous under some circumstancs.

Reply:
You're right, I was tripped up by the inconsistent patterns in this function. I made them more consistent in 504b16fb757d14f58e20294f7aca338dc6826140.